v2/spanner-to-sourcedb/src/main/java/com/google/cloud/teleport/v2/templates/dbutils/dml/CassandraTypeHandler.java

```diff
@@ -0,0 +1,974 @@
+   * @param formatter - the date format pattern used to parse date strings.
+   * @return a {@link LocalDate} object parsed from the given value.
+   * @throws IllegalArgumentException if the value cannot be parsed or is of an unsupported type.
+   */
+  public static LocalDate parseDate(String colName, Object colValue, String formatter) {
+    LocalDate localDate;
+    if (colValue instanceof String) {
+      try {
+        DateTimeFormatter dateFormatter = DateTimeFormatter.ofPattern(formatter);
+        localDate = LocalDate.parse((String) colValue, dateFormatter);
+      } catch (DateTimeParseException e) {
+        throw new IllegalArgumentException("Invalid date format for column " + colName, e);
+      }
+    } else if (colValue instanceof java.util.Date) {
+      localDate =
+          ((java.util.Date) colValue)
+              .toInstant()
+              .atZone(java.time.ZoneId.systemDefault())
+              .toLocalDate();
+    } else if (colValue instanceof Long) {
+      localDate =
+          java.time.Instant.ofEpochMilli((Long) colValue)
+              .atZone(java.time.ZoneId.systemDefault())
+              .toLocalDate();
+    } else {
+      throw new IllegalArgumentException("Unsupported type for column " + colName);
+    }
+    return localDate;
+  }
+
+  /**
+   * Generates a Type based on the provided {@link CassandraTypeHandler}.
+   *
+   * @param colName - which is used to fetch Key from valueJSON.
+   * @param valuesJson - contains all the key value for current incoming stream.
+   * @return a {@link String} object containing String as value represented in cassandra type.
+   */
+  public static String handleCassandraTextType(String colName, JSONObject valuesJson) {
+    return valuesJson.optString(
+        colName, null); // Get the value or null if the key is not found or the value is null
+  }
+
+  /**
+   * Generates a Type based on the provided {@link CassandraTypeHandler}.
+   *
+   * @param colName - which is used to fetch Key from valueJSON.
+   * @param valuesJson - contains all the key value for current incoming stream.
+   * @return a {@link UUID} object containing UUID as value represented in cassandra type.
+   */
+  public static UUID handleCassandraUuidType(String colName, JSONObject valuesJson) {
+    String uuidString =
+        valuesJson.optString(
+            colName, null); // Get the value or null if the key is not found or the value is null
+
+    if (uuidString == null) {
+      return null;
+    }
+
+    return UUID.fromString(uuidString);
+  }
+
+  /**
+   * Generates a Type based on the provided {@link CassandraTypeHandler}.
+   *
+   * @param colName - which is used to fetch Key from valueJSON.
+   * @param valuesJson - contains all the key value for current incoming stream.
+   * @return a {@link Long} object containing Long as value represented in cassandra type.
+   */
+  public static Long handleCassandraBigintType(String colName, JSONObject valuesJson) {
```

Comment:
It should be of type BigInteger: https://docs.datastax.com/en/drivers/java/4.1/com/datastax/oss/driver/api/core/data/SettableByName.html#setBigInteger-java.lang.String-java.math.BigInteger-

Reply:
@shreyakhajanchi  As per the design document, we need to convert int64 values to bigint.
In Cassandra, the bigint type corresponds to a 64-bit signed integer, which can be handled in Java using the Long type.
Therefore, I have used Long to represent the int64 values, as Long is the Java equivalent for handling 64-bit integer values.
<meta charset="utf-8"><b style="font-weight:normal;" id="docs-internal-guid-66f8c448-7fff-00fd-3956-ce24a3329bd4"><h4 dir="ltr" style="line-height:1.38;background-color:#ffffff;margin-top:0pt;margin-bottom:4pt;padding:10pt 0pt 0pt 0pt;"></h4><div dir="ltr" style="margin-left:0pt;" align="left">
Spanner Type | Cassandra Type
-- | --
Int64 | Bigint

</div></b>

I guess it will become more clear with DML PR

Reply:
For this particular type, `setLong` works too
[SettableByName.html#setLong-java.lang.String-long-](https://docs.datastax.com/en/drivers/java/4.1/com/datastax/oss/driver/api/core/data/SettableByName.html#setLong-java.lang.String-long-)